src/controllers/Talk_commentsController.php

```diff
@@ -130,4 +130,44 @@ public function moderateReportedComment($request, $db)
         header("Location: $uri", true, 204);
         exit;
     }
+
+    public function updateComment($request, $db)
+    {
+        $request->user_id = 14;
+
+        // must be logged in
+        if (!isset($request->user_id) || empty($request->user_id)) {
+            throw new Exception('You must log in to edit a comment', 401);
+        }
+
+        $new_comment_body = $request->getParameter('comment');
+        if (empty($new_comment_body)) {
+            throw new Exception('The field "comment" is required', 400);
+        }
+
+        $comment_id = $this->getItemId($request);
+        $comment_mapper = new TalkCommentMapper($db, $request);
+        $comment = $comment_mapper->getRawComment($comment_id);
+
+        if (false === $comment) {
+            throw new Exception('Comment not found', 404);
+        }
+
+        if ($comment['user_id'] != $request->user_id) {
+            throw new Exception('You are not the comment author', 401);
+        }
+
+        $max_comment_edit_minutes = 15;
+        if (isset($this->config['limits']['max_comment_edit_minutes'])) {
+            $max_comment_edit_minutes = $this->config['limits']['max_comment_edit_minutes'];
+        }
+
+        if ($comment['date_made'] + ($max_comment_edit_minutes * 60) < time()) {
+            throw new Exception('Cannot edit the comment after ' . $max_comment_edit_minutes . ' minutes', 400);
```

Comment:
I'm torn here - technically a 403 would satisfy (in the same way as a 403 for trying to access a directory listing that is 'forbidden for all'), but it doesn't _feel_ right to me? I think 400 is ok, or 405, because PUT isn't allowed after the 15 minute window?